server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -290,6 +287,11 @@ public String toString() {
     protected static ConfigKey<Boolean> vmStatsCollectUserVMOnly = new ConfigKey<>("Advanced", Boolean.class, "vm.stats.user.vm.only", "false",
             "When set to 'false' stats for system VMs will be collected otherwise stats collection will be done only for user VMs", true);
 
+    protected static ConfigKey<Long> vmStatsRemoveBatchSize = new ConfigKey<>("Advanced", Long.class, "vm.stats.remove.batch.size", "0", "Indicates the" +
```

Comment:
Should we have a generic global setting which we can use for other operations as well?
We can also reuse `detail.batch.query.size` for deletion as well.

Reply:
We could have a generic configuration for other operations, that is a good idea, however, this would incur in a lot more changes that escape this PR's scope. Furthermore, I don't think that we should reuse `detail.batch.query.size` for deletion, it is better to have more granularity on these configurations.